docs/testing.md

```diff
@@ -0,0 +1,338 @@
+# Testing Guide for insights-ros-ingress
+
+This document provides comprehensive testing instructions for the insights-ros-ingress service, including integration tests with docker-compose services.
+
+## Overview
+
+The testing setup validates the complete end-to-end flow:
+1. **Docker-compose services**: MinIO, Kafka, Zookeeper
+2. **Local service**: insights-ros-ingress running with test configuration
+3. **Upload API**: Accepts HCCM payloads with ROS data
+4. **Processing**: Extracts ROS files and uploads to MinIO
+5. **Messaging**: Sends Kafka events to ROS topic
+
+## Quick Start
+
+### 1. Run Full Integration Test
+```bash
+# Complete end-to-end test (starts services, tests, and cleans up)
+make test-integration
+```
+
+### 2. Manual Testing with Running Services
+```bash
+# Start docker-compose services
+make dev-env-up
+
+# Wait for services to be ready (30-60 seconds)
+sleep 30
+
+# Run the service with test configuration
+make run-test
+
+# In another terminal, test the upload API
+curl -X POST \
+  -H "Content-Type: multipart/form-data" \
+  -H "x-rh-identity: $(echo '{"identity":{"account_number":"12345","org_id":"12345","type":"User"}}' | base64 -w 0)" \
```

Comment:
I guess this will eventually need to be migrated to oauth 2 since we aren't supposed to support x-rh-identity. I can take a look later once this PR is merged to be sure that everything is align